hkmc2/shared/src/main/scala/hkmc2/semantics/Term.scala

```diff
@@ -67,11 +109,14 @@ enum Term extends Statement:
   case Handle(lhs: LocalSymbol, rhs: Term, args: List[Term],
     derivedClsSym: ClassSymbol, defs: Ls[HandlerTermDefinition], body: Term)
   
-  lazy val symbol: Opt[Symbol] = this match
+  def symbol: Opt[Symbol] = this match
+    case ref: Ref if ref.resSym.nonEmpty => ref.resSym
```

Comment:
I suppose this `symbol` method is returning the symbol of the thing that the term evaluates to. So I changed Ref's symbol to the new field `resSym`, which will be resolved by the resolver. If the resolver hasn't yet resolved the symbol or the resolver cannot resolve the new symbol, it falls back to the `sym` field of `Ref`

Reply:
I tried, and it seemed that we couldn't just make it crash (or even make it return `None`) at the time because so many places are using this `symbol` method before elaboration. I think the proper way is to clearly separate the preliminary symbols and the symbols after resolution because even in the resolution stage we need the symbols to determine what to do next...

Reply:
I agree, we should clearly separate the different kinds of symbols available at different times. But really, after we move all resolution to after elaboration, the only symbols we should really be looking at before resolution are the original symbols stored in Ref's main parameter list, no?

So, in an case, we shouldn't use a `symbol` method like the one you're defining here whose semantics _changes_ over the course of compilation.

The original `symbol` method was just a shorthand to check if a given term had a symbol (mostly, if it was a Ref) without having to perform pattern matching. Let's restore it to that use case.